libmamba/CMakeLists.txt

```diff
@@ -609,6 +609,8 @@ macro(libmamba_create_target target_name linkage output_name)
                 solv::libsolvext
                 solv::cpp
         )
+        # CMake 3.17 provides a LibArchive::LibArchive target
```

Comment:
It's always been required, you just happen to always install `archive.h` into a location that is shared with other headers, so that an unrelated include picks it up. In brew, it's installed to a separate location so that it doesn't interfere with the system, so it matters.

Before CMake 3.17, there is no `LibArchive::LibArchive` target. This note is for whenever 3.17 can be made the minimum.

Unrelated, but you should never use `include_directories` in CMake. You should always use `target_include_directories` instead.

Reply:
I reworded, is that better?

Reply:
Also, this is probably true for the other things that are included as `_LIBRARIES` too, like Curl. I only fixed the one that was clearly broken. In general, I recommend always using targets, making them by hand if they are not provided by find_package. In Curl's case, `CURL::libcurl` was [added in 3.12](https://cmake.org/cmake/help/latest/module/FindCURL.html), so you should use that.

Reply:
Yes thank you for the comprehensive comment.

Feel free to improve the build systems of mamba for the better if you are interested.

Reply:
I'd rather provide advice and have other people fix it. :P

Reply:
(I might come back and improve things in the future, but limited on time till Dec 2 or so)